diff --git nova/network/l3.py nova/network/l3.py
index 4603bdf..b29c33a 100644
--- nova/network/l3.py
+++ nova/network/l3.py
@@ -121,6 +121,12 @@ class LinuxNetL3(L3Driver):
         pass
 
 
+class LinuxNetNoNatL3(LinuxNetL3):
+    """L3 driver that uses linux_net as the backend without ahy SNAT"""
+    def initialize_network(self, cidr):
+        pass
+
+
 class NullL3(L3Driver):
     """The L3 driver that doesn't do anything.  This class can be used when
        nova-network shuld not manipulate L3 forwarding at all (e.g., in a Flat
diff --git nova/network/quantum/melange_ipam_lib.py nova/network/quantum/melange_ipam_lib.py
index c68d83c..bb1755d 100644
--- nova/network/quantum/melange_ipam_lib.py
+++ nova/network/quantum/melange_ipam_lib.py
@@ -152,7 +152,7 @@ class QuantumMelangeIPAMLib(object):
 
     def get_tenant_id_by_net_id(self, context, net_id, vif_id, project_id):
         ipam_tenant_id = None
-        tenant_ids = [FLAGS.quantum_default_tenant_id, project_id, None]
+        tenant_ids = [project_id, FLAGS.quantum_default_tenant_id, None]
         # This is confusing, if there are IPs for the given net, vif,
         # tenant trifecta we assume that is the tenant for that network
         for tid in tenant_ids:
